src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,207 @@
+    {
+        private readonly static Metadata Metadata = new("Flagsmith Provider");
+        internal readonly IFlagsmithClient _flagsmithClient;
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        public FlagsmithProvider(IFlagsmithConfiguration options)
+        {
+            _flagsmithClient = new FlagsmithClient(options);
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        /// <param name="httpClient">Http client that will be used for flagsmith requests. You also can use it to register <see cref="FeatureProvider"/> as Typed HttpClient with <see cref="FeatureProvider"> as abstraction</see></param>
+        public FlagsmithProvider(IFlagsmithConfiguration options, HttpClient httpClient)
+        {
+            _flagsmithClient = new FlagsmithClient(options, httpClient);
+        }
+
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="flagsmithClient">Precreated Flagsmith client. You can just use <see cref="FlagsmithClient"/> class.</param>
+        public FlagsmithProvider(IFlagsmithClient flagsmithClient)
+        {
+            _flagsmithClient = flagsmithClient;
+        }
+
+        private Task<IFlags> GetFlags(EvaluationContext ctx = null)
+        {
+            var key = ctx?.GetValue("targetingKey")?.AsString;
+            return string.IsNullOrEmpty(key)
+                ? _flagsmithClient.GetEnvironmentFlags()
+                : _flagsmithClient.GetIdentityFlags(key, ctx.AsDictionary().Select(x => new Trait(x.Key, x.Value.AsObject) as ITrait).ToList());
+        }
+
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata() => Metadata;
+
+        /// <inheritdoc/>
+
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            var flags = await GetFlags(context);
+            var isFlagEnabled = await flags.IsFeatureEnabled(flagKey);
+            if (!isFlagEnabled)
+            {
+                return new ResolutionDetails<bool>(flagKey, defaultValue, reason: Reason.Disabled);
+            }
+
+            var stringValue = await flags.GetFeatureValue(flagKey);
+            if (bool.TryParse(stringValue, out var parsedValue))
+            {
+                return new ResolutionDetails<bool>(flagKey, parsedValue);
+            }
+            return new ResolutionDetails<bool>(flagKey, defaultValue, ErrorType.ParseError, Reason.Error, errorMessage: "Failed to parse value in int type");
```

Comment:
You might be able to use generics to reduce some duplication here - one function to do the error changes and type conversion, but that's completely up to you.

Reply:
I tried to do it, but didn't found a good solution for reusing TryParse, because these types doesn't have an interface for this...

Reply:
You could try to extract `TryParse` into a lambda which you pass to the single generic function, but again this is up to you. The duplication isn't that bad IMHO.

Reply:
done :)

Reply:
Nice!